backend/app/routers/journal.py

```
@@ -132,6 +147,62 @@ def import_transactions(req: ImportRequest, db: DBSession = Depends(get_db)):
+    """Preview options trades from a Schwab transaction CSV export.
+
+    Mirrors the Schwab API preview path but reads from an uploaded CSV file.
+    Use this when historical transactions predate the developer-app creation
+    date and are therefore not returned by ``/api/journal/import/preview``.
+    """
+    file_bytes = await _read_csv_upload(file)
+    try:
+        mapped = parse_schwab_csv(file_bytes)
+    except Exception:
+        logger.exception("Failed to parse uploaded Schwab CSV")
+        raise HTTPException(status_code=422, detail=_CSV_PARSE_DETAIL)
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)